### PR TITLE
feat: add postgres database driver

### PR DESCRIPTION
**Pull request by codekeyz**
## Description

This adds support for using Yaroorm with a PostgresSQL Database.

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [x] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [x] 🗑️ Chore

### CODE REVIEW

File: yaroorm/lib/migration.dart [hunk 1]
```diff
@@ -5,6 +5,10 @@ import 'package:recase/recase.dart';
 
 import 'src/database/entity.dart';
 
+class Id {}
+
+enum Integer { smallint, integer, bigint }
+
```
Comment (codekeyz) [inline yaroorm/lib/migration.dart:11]: I don't see these being used. Let's remove it

File: yaroorm/lib/src/database/driver/pgsql_driver.dart [hunk 2]
```diff
@@ -0,0 +1,279 @@
+import 'package:meta/meta.dart';
+import 'package:postgres/postgres.dart' as pg;
+import 'package:yaroorm/migration.dart';
+import 'package:yaroorm/src/database/driver/mysql_driver.dart';
+import 'package:yaroorm/src/query/primitives/serializer.dart';
+import 'package:yaroorm/yaroorm.dart';
+
+import 'sqlite_driver.dart' show SqliteSerializer;
+
+final _primitiveSerializer = PgSqlPrimitiveSerializer();
```
Comment (codekeyz) [inline yaroorm/lib/src/database/driver/pgsql_driver.dart:10]: Why do we have two definitions of this? See line `L18`?

File: yaroorm/lib/src/database/driver/pgsql_driver.dart [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+
+  @override
+  Future execute(String script) => rawQuery(script);
+
+  @override
+  Future<int> insert(InsertQuery query) async {
+    // if (!isOpen) await connect();
+    final sql = _primitiveSerializer.acceptInsertQuery(query);
+    final result = await db.execute(sql);
+    return result.affectedRows;
```
Comment (codekeyz) [inline yaroorm/lib/src/database/driver/pgsql_driver.dart:68]: I have doubts about this code. An insert should return the record ID, not affected rows

File: yaroorm/lib/src/database/driver/pgsql_driver.dart [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+import 'package:yaroorm/src/query/primitives/serializer.dart';
+import 'package:yaroorm/yaroorm.dart';
+
+import 'sqlite_driver.dart' show SqliteSerializer;
+
+final _primitiveSerializer = PgSqlPrimitiveSerializer();
+
+class PostgreSqlDriver implements DatabaseDriver {
+  final DatabaseConnection config;
+  late pg.Connection db;
```
Comment (codekeyz) [inline yaroorm/lib/src/database/driver/pgsql_driver.dart:14]: This shouldn't be marked as `late`. It should be nullable.

File: yaroorm/lib/src/database/driver/pgsql_driver.dart [hunk 5]
```diff
@@ -0,0 +1,313 @@
...
+    db = await pg.Connection.open(
+        pg.Endpoint(
+          host: config.host!,
+          database: config.database,
+          username: config.username,
+          password: config.password,
+          port: config.port == null ? 5432 : config.port!,
+        ),
+        settings: pg.ConnectionSettings(sslMode: secure ? pg.SslMode.require : pg.SslMode.disable, timeZone: 'GMT'));
+    return this;
```
Comment (codekeyz) [inline yaroorm/lib/src/database/driver/pgsql_driver.dart:37]: Is there any reason we're hardcoding timezone here?

File: yaroorm/lib/src/database/driver/pgsql_driver.dart [hunk 6]
```diff
@@ -0,0 +1,313 @@
...
+  @override
+  void binary(String name,
+      {bool nullable = false, String? defaultValue, String? charset, String? collate, int size = 1}) {
+    statements.add(_getColumn(name, "BYTEA", nullable: nullable, defaultValue: null));
+  }
+
+  @override
+  void varbinary(String name,
+      {bool nullable = false, String? defaultValue, String? charset, String? collate, int size = 1}) {
+    throw UnimplementedError('varbinary not implemented for Postgres');
```
Comment (codekeyz) [inline yaroorm/lib/src/database/driver/pgsql_driver.dart:299]: Postgres doesn't support this type?

File: yaroorm/lib/src/database/driver/pgsql_driver.dart [hunk 7]
```diff
@@ -0,0 +1,313 @@
...
+  }
+
+  @override
+  void enums(String name, List<String> values,
+      {bool nullable = false, String? defaultValue, String? charset, String? collate}) {
+    throw UnimplementedError('enums not implemented for Postgres');
```
Comment (codekeyz) [inline yaroorm/lib/src/database/driver/pgsql_driver.dart:305]: Also enums?

File: yaroorm/lib/src/database/driver/pgsql_driver.dart [hunk 8]
```diff
@@ -0,0 +1,313 @@
...
+  }
+
+  @override
+  void set(String name, List<String> values,
+      {bool nullable = false, String? defaultValue, String? charset, String? collate}) {
+    throw UnimplementedError('set not implemented for Postgres');
```
Comment (codekeyz) [inline yaroorm/lib/src/database/driver/pgsql_driver.dart:311]: And lastly this? @donkizzy

Comment (donkizzy) [reply yaroorm/lib/src/database/driver/pgsql_driver.dart:310]: Postgres doesn't support SET 

there are workarounds but that's an entire different thing and not the same thing

https://stackoverflow.com/questions/14674461/mysql-set-type-in-postgresql

**Comment by codekeyz** [comment]
@donkizzy Finally, we're set to close this PR. Postgres Driver has passed all the tests 🔥🚀. Thanks for the work on this.

**Pull request merged**